Makefile

```diff
@@ -455,23 +455,25 @@ ifneq ($(wildcard $(MAGMA_DIR)/lib/libmagma.*),)
     endif
   else  # HIP MAGMA
     ifneq ($(HIP_LIB_DIR),)
-      hip_link = $(if $(STATIC),,-Wl,-rpath,$(HIP_LIB_DIR)) -L$(HIP_LIB_DIR) -lhipblas -lhipsparse -lamdhip64
+      HIP_LIB_DIR_MAGMA=$(HIP_DIR)/../lib
+      HIP_INC_DIR_MAGMA=$(HIP_DIR)/../include
```

Comment:
I'm confused why this change is needed.  What were you using as `HIP_DIR`?  I generally use `/opt/rocm` for `HIP_DIR` -- perhaps I should have named it `ROCM_DIR` -- but with my usual build configuration, I get `HIP_LIB_DIR_MAGMA` as `/opt/rocm/../lib` which is not what I want.

Reply:
My `HIP_DIR` points to the "hip dir" :), which is `/opt/rocm-xx/hip`. I got confused that it is not named `ROCM_DIR` or `ROCM_PATH` (the latter is declared by default when load a rocm module on Crusher).

Reply:
Should we change the name?  How many other people's build systems could that interfere with?

I was wondering how you could get all the backends to build at the same time (the "regular" HIP backends and MAGMA), but I see at least some ROCm installs have the basic libraries we need in `/opt/rocm/hip/lib` as well as `/opt/rocm/lib`.  So the current setup will work to build all the backends with `HIP_DIR` as `/opt/rocm/hip`, but only the "standard" HIP backends with `HIP_DIR = /opt/rocm`.

Are there any strong opinions about this?  In the README, we do say "To enable HIP support, add `HIP_DIR=/opt/rocm` or an appropriate directory.", but that could be changed.

Reply:
I think we should change the makefile to use `ROCM_DIR` to the extent we need that path (e.g., as default for `hipcc`). We're going to need to revisit the HIP build soon to remove AMD-specific assumptions because the CHIP-SPV project is almost usable by libCEED, and will open one (of two) paths to Intel GPUs. The details block below is how one developer hacked the makefile to build with CHIP-SPV, in case that's useful when updating it.

<details>

```diff
diff --git a/Makefile b/Makefile
index 54c99769..7d34364c 100644
--- a/Makefile
+++ b/Makefile
@@ -397,16 +397,21 @@ ifneq ($(CUDA_LIB_DIR),)
 endif

 # HIP Backends
-HIP_LIB_DIR := $(wildcard $(foreach d,lib
lib64,$(HIP_DIR)/$d/libamdhip64.${SO_EXT}))
-HIP_LIB_DIR := $(patsubst %/,%,$(dir $(firstword $(HIP_LIB_DIR))))
+# Probably should extend to look for libCHIP.so?
+# HIP_LIB_DIR := $(wildcard $(foreach d,lib
lib64,$(HIP_DIR)/$d/libamdhip64.${SO_EXT}))
+HIP_LIB_DIR :=
/gpfs/jlse-fs0/users/pvelesko/install/HIP/clang15/chip-spv-develop-debug/lib
 HIP_BACKENDS = /gpu/hip/ref /gpu/hip/shared /gpu/hip/gen
 ifneq ($(HIP_LIB_DIR),)
   $(libceeds) : HIPCCFLAGS += -I./include
   ifneq ($(CXX), $(HIPCC))
-    CPPFLAGS += $(subst =,,$(shell $(HIP_DIR)/bin/hipconfig -C))
+    # removing the equal signs breaks options such as
--target=x86_64-linux-gnu
+    # CPPFLAGS += $(subst =,,$(shell $(HIP_DIR)/bin/hipconfig -C))
+    CPPFLAGS += $(shell $(HIP_DIR)/bin/hipconfig -C)
   endif
   $(libceeds) : CPPFLAGS += -I$(HIP_DIR)/include
-  PKG_LIBS += -L$(abspath $(HIP_LIB_DIR)) -lamdhip64 -lhipblas
+  # extracted from hipconfig
+  PKG_LIBS +=
-L/home/pvelesko/space/install/HIP/clang15/chip-spv-testing/lib -lCHIP
-L/soft/libraries/khronos/loader/master-2022.05.18/lib64 -lOpenCL
-L/soft/restricted/CNDA/emb/intel-gpu-umd/20221031.1-pvc-prq-66/driver/lib64
-lze_loader
-Wl,-rpath,/home/pvelesko/space/install/HIP/clang15/chip-spv-testing/lib
+#   PKG_LIBS += -L$(abspath $(HIP_LIB_DIR)) -lamdhip64 -lhipblas
   LIBCEED_CONTAINS_CXX = 1
   libceed.c     += interface/ceed-hip.c
   libceed.c     += $(hip.c) $(hip-ref.c) $(hip-shared.c) $(hip-gen.c)
@@ -490,7 +495,7 @@ $(libceed.a) : $(call weak_last,$(libceed.o)) |
$$(@D)/.DIR
        $(call quiet,AR) $(ARFLAGS) $@ $^

 $(OBJDIR)/%.o : $(CURDIR)/%.c | $$(@D)/.DIR
-       $(call quiet,CC) $(CPPFLAGS) $(CFLAGS) -c -o $@ $(abspath $<)
+       $(call quiet,CC) $(CFLAGS) -c -o $@ $(abspath $<)

 $(OBJDIR)/%.o : $(CURDIR)/%.cpp | $$(@D)/.DIR
        $(call quiet,CXX) $(CPPFLAGS) $(CXXFLAGS) -c -o $@ $(abspath $<)
@@ -687,7 +692,7 @@ style : style-c style-py
 CLANG_TIDY ?= clang-tidy

 %.c.tidy : %.c
-       $(CLANG_TIDY) $(TIDY_OPTS) $^ -- $(CPPFLAGS) --std=c99
-I$(CUDA_DIR)/include -I$(HIP_DIR)/include
-DCEED_JIT_SOUCE_ROOT_DEFAULT="\"$(abspath ./include)/\""
+       $(CLANG_TIDY) $(TIDY_OPTS) $^ -- --std=c99 -I$(CUDA_DIR)/include
-I$(HIP_DIR)/include -DCEED_JIT_SOUCE_ROOT_DEFAULT="\"$(abspath
./include)/\""

 %.cpp.tidy : %.cpp
        $(CLANG_TIDY) $(TIDY_OPTS) $^ -- $(CPPFLAGS) --std=c++11
-I$(CUDA_DIR)/include -I$(OCCA_DIR)/include -I$(HIP_DIR)/include
```

</details>

Reply:
Please check out 5e5b0c9.